File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/JAVA_TEMA/tema_bun/Server/src/main/java/com/example/tema_bun/StartRpcServer.java
package com.example.tema_bun;

import com.example.tema_bun.servers.AbstractServer;
import com.example.tema_bun.servers.RpcConcurrentServer;

import javax.swing.*;
import java.io.FileReader;
import java.io.IOException;
import java.util.Properties;

public class StartRpcServer {
    private static int defaultPort=55555;

    public static void main(String[] args){
        Properties serverProps=new Properties();
        try {
//            serverProps.load(new FileReader("D:\\FACULTATE\\AN 2\\SEMESTRUL IV\\MPP\\LAB_3\\mpp-proiect-java-aabgx\\tema_bun\\Server\\src\\main\\resources\\server.properties"));
            serverProps.load(new FileReader("D:\\FACULTATE\\AN 2\\SEMESTRUL IV\\MPP\\JAVA_CSHARP_TEMA\\mpp-proiect-java-aabgx\\tema_bun\\Server\\src\\main\\resources\\server.properties"));
            System.out.println("Server properties set. ");
            serverProps.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find chatserver.properties "+e);
            return;
        }
        UtilizatorRepository utilizatorRepository=new UtilizatorRepository(serverProps);
        CopilRepository copilRepository=new CopilRepository(serverProps);
        ProbaRepository probaRepository=new ProbaRepository(serverProps);
        InscriereRepository inscriereRepository=new InscriereRepository(serverProps);


        Service service = new Service(utilizatorRepository,copilRepository,probaRepository,inscriereRepository);
        int serverPort = defaultPort;
        try {
            serverPort = Integer.parseInt(serverProps.getProperty("server.port"));
        } catch (NumberFormatException nef) {
            System.err.println("Wrong Port Number" + nef.getMessage());
            System.err.println("Using default port " + defaultPort);
        }
        System.out.println("Starting server on port: " + serverPort);
        AbstractServer server = new RpcConcurrentServer(serverPort, service);
        try {
            server.start();
        } catch (Exception e) {
            System.err.println("Error starting the server" + e.getMessage());
        } finally {
            try {
                server.stop();
            } catch (Exception e) {
                System.err.println("Error stopping server " + e.getMessage());
            }
        }
    }
}

File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/CSHARP_TEMA/tema_sapt3/Server/StartRpcServer.cs
using log4net;
using services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Mime;
using System.Text;
using System.Threading.Tasks;
using networking.servers;
using persistance;
using Microsoft.EntityFrameworkCore;

namespace Server
{
    public class StartRpcServer
    {
        private static int defaultPort = 55555;

        static void Main(string[] args)
        {
            log4net.Config.XmlConfigurator.ConfigureAndWatch(new System.IO.FileInfo(AppDomain.CurrentDomain.BaseDirectory + "App.config"));
            
            /*Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);*/
            
            /*var builder = new ConfigurationBuilder()
                .SetBasePath(Directory.GetCurrentDirectory())
                .AddJsonFile("C:\\Users\\Marius Andreiasi\\RiderProjects\\concursInot\\clientWF\\config.json", optional: true, reloadOnChange: true);
            IConfigurationRoot configuration = builder.Build();*/
            /*string connectionString = configuration.GetConnectionString("MyConnectionString");*/
            //Console.WriteLine(connectionString);
            /*log4net.Config.XmlConfigurator.Configure();*/

            /*ILog logger1 = LogManager.GetLogger("CopilRepository");*/
            CopilRepository copilRepository = new CopilRepository();

            /*ILog logger2 = LogManager.GetLogger("InscriereRepository");*/
            InscriereRepository inscriereRepository = new InscriereRepository();

            /*ILog logger3 = LogManager.GetLogger("ProbaRepository");*/
            ProbaRepository probaRepository = new ProbaRepository();
            
            /*ILog logger4 = LogManager.GetLogger("UtilizatorRepository");*/
            UtilizatorRepository utilizatorRepository = new UtilizatorRepository();

            var contextOptions  = new DbContextOptionsBuilder<DBContext>()
                .UseSqlite("Data Source=D:\\FACULTATE\\AN 2\\SEMESTRUL IV\\MPP\\JAVA_CSHARP_TEMA\\mpp-proiect-java-aabgx\\tema_bun\\concurs.db")
                .Options;
            //optionBuilder.UseSqlite(@"Data Source=D:\\FACULTATE\\AN 2\\SEMESTRUL IV\\MPP\\JAVA_CSHARP_TEMA\\mpp-proiect-java-aabgx\\tema_bun\\concurs.db");
            var context = new DBContext(contextOptions );

            //IServices Service = new Service(copilRepository, inscriereRepository, probaRepository,utilizatorRepository);
            IServices Service = new Service(context);
            
            int serverPort = defaultPort;
            AbstractServer server = new RpcConcurrentServer(serverPort, Service);
            try
            {
                server.start();
            }
            catch (ServerException e)
            {
                Console.Error.WriteLine("Error starting the server" + e.Message);
            }
            finally
            {
                try
                {
                    server.stop();
                }
                catch (ServerException e)
                {
                    Console.Error.WriteLine("Error stopping server " + e.Message);
                }
            }
        }
    }
}
